File: 백준/Silver/11047. 동전 0/동전 0.py
n, k = map(int, input().split())

coins = []

for i in range(n):
    coins.append(int(input()))

coins.sort(reverse = True)

count = 0

for coin in coins:
    if(k >= coin):
        count += k // coin
        k = k % coin

print(count)

File: 백준/Silver/11047. 동전 0/동전 0.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let [n, k] = input[0].split(" ").map(Number);
let arr = [];

for(let i = 1; i <= n; i++){
  arr.push(Number(input[i]));
}
let answer = 0;

arr.sort((a, b) => b - a);

for(let value of arr){
  answer += parseInt(k / value);
  k = k % value;
}
console.log(answer);